examples/js/effects/VREffect.js

```diff
@@ -55,34 +69,76 @@ THREE.VREffect = function ( renderer, onError ) {
 
 	// fullscreen
 
-	var isFullscreen = false;
+	var isPresenting = false;
 
 	var canvas = renderer.domElement;
 	var fullscreenchange = canvas.mozRequestFullScreen ? 'mozfullscreenchange' : 'webkitfullscreenchange';
 
-	document.addEventListener( fullscreenchange, function ( event ) {
+	document.addEventListener( fullscreenchange, function () {
+
+		if ( deprecatedAPI ) {
+
+			isPresenting = document.mozFullScreenElement || document.webkitFullscreenElement;
+
+		}
+
+	}, false );
+
+	window.addEventListener( 'vrdisplaypresentchange', function () {
 
-		isFullscreen = document.mozFullScreenElement || document.webkitFullscreenElement;
+		if (vrHMD) {
+
+			isPresenting = vrHMD.isPresenting;
+
+		}
 
 	}, false );
 
 	this.setFullScreen = function ( boolean ) {
 
 		if ( vrHMD === undefined ) return;
-		if ( isFullscreen === boolean ) return;
+		if ( isPresenting === boolean ) return;
+
+		if ( !deprecatedAPI ) {
+
+			if ( boolean ) {
+
+				vrHMD.requestPresent( { source: canvas } );
+
+			} else {
+
+				vrHMD.exitPresent();
+
+			}
+
+		} else {
+
+			if ( canvas.mozRequestFullScreen ) {
 
-		if ( canvas.mozRequestFullScreen ) {
+				canvas.mozRequestFullScreen( { vrDisplay: vrHMD } );
 
-			canvas.mozRequestFullScreen( { vrDisplay: vrHMD } );
+			} else if ( canvas.webkitRequestFullscreen ) {
 
-		} else if ( canvas.webkitRequestFullscreen ) {
+				canvas.webkitRequestFullscreen( { vrDisplay: vrHMD } );
 
-			canvas.webkitRequestFullscreen( { vrDisplay: vrHMD } );
+			} else {
+
+				console.error( 'No compatible requestFullscreen method found.' );
```

Comment:
I wish this method (always) returned a Promise so we could `reject` here - would be useful for handling the Safari/iOS case, for example